src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -499,7 +578,7 @@ internal override ValueTask CloseAsync(long errorCode, CancellationToken cancell
 
         private void ThrowIfDisposed()
         {
-            if (_disposed)
+            if (_disposed == 1)
```

Comment:
Should this be a volatile read?

Reply:
I don't think it needs to. It can be disposed any time immediately after the check anyway. We would need much bigger change to make sure that we don't dispose while in msquic. I plan to follow up on MsQuciStream and I'll take a look at the connection as well.

Reply:
I think it actually is needed here, longer-term. Without volatile read (and write on other side) there is a visibility problem on ARM.